docs(website): add 'Getting Started' section (#2320)
rome/tools: website/src/_includes/docs/getting-started.md

## Getting Started

Install Rome via one of the following alternatives.

### Installation of the CLI

You can install the `rome` binary via any node package manager.

> > WARNING: we strongly suggest to **not** install the binary globally, instead install the binary

Reviewer: Maybe we could direct people that just want to try out the CLI to use `npx rome@next` instead of having to add it to their `package.json` upfront

Author: Does `yarn` have something like `npx`?

Author: `yarn` v2 has `yarn dlx`, it seems. I will add only `npx/pnpx` for now. I know the community is quite divided on yarn v1 and v2.

> locally to your project and create local commands in order to leverage the CLI.

> #### [Installation via Yarn](https://yarnpkg.com/)

Reviewer: It would be nice if the website / theme had a "tabbed widget" to collapse these kind of "platform specific" examples into a single section

Author: We could look into it later. We don't have such feature for now (it would require some HTML/CSS/JS work). At least, like this, each section has its own link.

Reviewer 2: It feels a bit excessive to have a new section for each manager. How about a table? 

```md
| Package Manager | Instructions | 
| npm | npm i rome@next |
| pnpm | pnpm i rome@next |
| yarn| yarn add rome@next |
```

### Install Rome CLI

Install `rome` using your preferred node package manager.

> RECOMMENDATION: we advise to **not** install the binary globally, instead install the binary
> locally to your project or use `npx/pnpx/yarn dlx` in order to leverage the CLI.

| Package manager               | Instructions         |
|-------------------------------|----------------------|
| [npm](https://www.npmjs.com/) | `npm i rome@next`    |
| [pnpm](https://pnpm.io/)      | `pnpm i rome@next`   |
| [yarn](https://yarnpkg.com/)  | `yarn add rome@next` |

### Install Rome in your CI pipeline

#### GitHub Actions

Please refer to the documentation of the [`setup-rome` action](https://github.com/rome/setup-rome#usage) for more information about its configuration: 

```yaml
steps:
    - uses: rome/setup-rome@v0.1
      with:
        version: latest 
    - run: rome --help
```

#### Installation on any other CI

You can download and install the binary directly using `curl`:

```shell
curl -L https://github.com/rome/tools/releases/download/latest/rome-<OS>-<ARCH> -o rome
chmod +x rome
```

> Please make sure to choose the correct architecture from the [releases page](https://github.com/rome/tools/releases).

Reviewer: We could expand on that by adding the "supported platforms" table here, explaining that `<OS>` and `<ARCH>` follow the node.js / VScode syntax, and the accepted values are `win32`, `darwin` and `linux` for `<OS>`, and `x64` and `arm64` for `<ARCH>` (this is different from the "target triple" format used by LLVM and Rust for example where `x64` is called `x86_64` and `win32` is called `windows`)

Author: Please check the new section and if it's how you imagined. I would have added a link to every single distribution but their link has the version number. If we had something like `https://github.com/rome/tools/releases/download/latest/rome_lsp-darwin-arm64.vsix` (notice the `/latest/` instead of the version number), we could point to them to a link.

Reviewer 2: You can use latest to get the latest release: `https://github.com/rome/tools/releases/latest/download/rome-linux-x64`

Author: > You can use latest to get the latest release: `https://github.com/rome/tools/releases/latest/download/rome-linux-x64`

Shoot! I saw this comment just now! I will fix it with another PR

## Getting Started

> Install Rome via one of the following alternatives.

Author: I can't find a section where, for example, `deno` mention that. (as they use Rust and they are bound to it for architecture support)

We could add a small paragraph saying that we don't support all the architectures and to check the release page? Not sure how phrase it.

> Install Rome via one of the following alternatives.

Reviewer 2: Excellent point. We could inspire our own getting started on how deno structures their guide. 

That's what I found at the top of [their guide](https://deno.land/manual/getting_started/installation.md0 and we could use a very similar copy saying that we support macOS, Linux, and Windows, both x64 and ARM64.

> Deno works on macOS, Linux, and Windows. Deno is a single binary executable. It has no external dependencies.

> On macOS, both M1 (arm64) and Intel (x64) executables are provided. On Linux and Windows, only x64 is supported.

> Install Rome via one of the following alternatives.

Author: I updated the section, let me know if it's fine

> ### Installation of the CLI

Reviewer 2: I would rephrase the title to be more scenario centered: 

```
Use Rome's CLI
```

Author: I wanted to keep a link that "talks", descriptive. What about `Install Rome's CLI`?

Author: And it's not actually "use", we are actually installing it... 🤔

You can install the `rome` binary via any node package manager.

> WARNING: we strongly suggest to **not** install the binary globally, instead install the binary
> locally to your project or use `npx/pnpx/yarn dlx` in order to leverage the CLI.

#### [Installation via Yarn](https://yarnpkg.com/)

```bash
yarn add rome@next
```

#### [Installation via npm](https://www.npmjs.com/)

```bash
npm i rome@next
```

#### [Installation via pnpm](https://pnpm.io/)

```bash
pnpm i rome@next
```

> ### Installation VS Code extension

Author: > Do we need to mention anything in case prettier is installed too and how they can make Rome the preferred formatter?

Good point! I think we explain that in the `formatter.md`, where we have the `### Use the formatter with the VSCode extension` section.

You can use Rome by downloading the [VS Code extension](https://marketplace.visualstudio.com/items?itemName=rome.rome) from the marketplace. 

> ### Installation via CI

Reviewer 2: Nit:
```suggestion
### Use Rome in your CI pipeline
```

> You can install Rome in a continues integration too.

Reviewer 2: I would leave out this sentence. It doesn't add more information than the sections following.

> #### Installation on GitHub

Reviewer 2: ```suggestion
#### GitHub Actions
```

> Please refer to the [official page](https://github.com/rome/setup-rome#usage) for more information about its configuration: 

Reviewer 2: ```suggestion
Please refer to the documentation of the [`setup-rome` action](https://github.com/rome/setup-rome#usage) for more information about its configuration: 
```

```yaml
steps:
    - uses: rome/setup-rome@v0.1
      with:
        version: latest 
    - run: rome --help
```

#### Installation on any other CI

You can download and install the binary directly using `curl`:

```shell

> curl -L https://github.com/rome/tools/releases/download/latest/rome-linux-x64 -o rome

Reviewer 2: ```suggestion
curl -L https://github.com/rome/tools/releases/download/latest/rome-<OS>-<ARCH> -o rome
```

chmod +x rome
```

Please make sure to choose the correct architecture from the [releases page](https://github.com/rome/tools/releases).

## After installation

- check the [formatter section](/formatter#use-the-formatter-with-the-cli) for options and commands;

> - check the options available in the [VS Code extension](/formatter#use-the-formatter-with-the-vscode-extension)

Reviewer 2: I would move this to the editor section. It isn't meaningful for someone that just installed the CI.

Author: The things is, once we installed Rome (regardless what), we need direct new users to somewhere. We can't assume that they know what to do/where to go after the installation. How would you guide new users?

## Getting Started

Rome works on Linux, macOS, and Windows operative systems. 

> On macOS, both M1 (arm64) and Intel (x64) executables are provided. On Linux and Windows, only x64 is supported.

Reviewer: Actually we do support ARM CPUs on Windows and Linux too

| Package manager               | Instructions         |
|-------------------------------|----------------------|
| [npm](https://www.npmjs.com/) | `npm i rome@next`    |
| [pnpm](https://pnpm.io/)      | `pnpm i rome@next`   |
| [yarn](https://yarnpkg.com/)  | `yarn add rome@next` |

### Install Rome in your CI pipeline

#### GitHub Actions

Please refer to the documentation of the [`setup-rome` action](https://github.com/rome/setup-rome#usage) for more information about its configuration: 

```yaml
steps:
    - uses: rome/setup-rome@v0.1
      with:
        version: latest 
    - run: rome --help
```

#### Installation on any other CI

You can download and install the binary directly using `curl`:

```shell
curl -L https://github.com/rome/tools/releases/download/latest/rome-<OS>-<ARCH> -o rome
chmod +x rome
```

Where `OS` and `ARCH` follow the Node.js syntax convention:

- `OS` is one of the following values: `win32`, `darwin` or `linux`
- `ARCH` is one of the following values: `arm64` or `x64`

|         | `win32`       | `darwin`       | `linux`       |
|---------|---------------|----------------|---------------|
| `arm64` | `win32-arm64` | `darwin-arm64` | `linux-arm64` | 

> | `x64`   | `win32-x64`   | `darwin-arm64` | `linux-arm64` | 

Reviewer: ```suggestion
| `x64`   | `win32-x64`   | `darwin-x64`   | `linux-x64`   | 
```
